cargo-sample/src/test/scala/io/cafienne/bounded/cargosample/httpapi/CargoRouteSpec.scala

```diff
@@ -0,0 +1,75 @@
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package io.cafienne.bounded.cargosample.httpapi
+
+import java.time.ZonedDateTime
+import java.util.UUID
+
+import akka.event.Logging
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import io.cafienne.bounded.aggregate.{AggregateRootCommand, CommandGateway, ValidateableCommand}
+import io.cafienne.bounded.cargosample.domain.CargoDomainProtocol
+import io.cafienne.bounded.cargosample.domain.CargoDomainProtocol.{CargoId, CargoNotFound}
+import io.cafienne.bounded.cargosample.projections.QueriesJsonProtocol.CargoViewItem
+import io.cafienne.bounded.cargosample.projections.{CargoQueries, QueriesJsonProtocol}
+import org.scalatest._
+
+import scala.concurrent.Future
+
+class CargoRouteSpec extends FlatSpec with MustMatchers with ScalatestRouteTest {
+
+  import spray.json._
+  import HttpJsonProtocol._
+  import io.cafienne.bounded.cargosample.domain.CargoDomainJsonProtocol._
+  import io.cafienne.bounded.aggregate.CommandEventDatastructureJsonProtocol._
+  import QueriesJsonProtocol._
+
+  val logger = Logging(system, getClass)
+
+
+  val cargoId1 = CargoId(UUID.fromString("8CD15DA4-006B-478C-8640-2FA52AA7657E"))
+  val cargoViewItem1 = CargoViewItem(cargoId1, "Amsterdam", "New York", ZonedDateTime.parse("2018-01-01T12:25:38.492+01:00[Europe/Amsterdam]"))
+
+  val cargoQueries = new CargoQueries {
+    override def getCargo(cargoId: CargoDomainProtocol.CargoId): Future[CargoViewItem] = {
+      if (cargoId.id.compareTo(cargoId1.id) == 0) {
+        Future.successful(cargoViewItem1)
+      } else {
+        Future.failed[CargoViewItem](CargoNotFound(s"Cargo with id $cargoId is not found"))
+      }
+    }
+  }
+
+  //TODO make use of testing command gateway ?
+  val commandGateway = new CommandGateway {
+    override def send[T <: AggregateRootCommand](command: T)(implicit validator: ValidateableCommand[T]): Future[Unit] = ???
+
+    override def sendAndAsk[T <: AggregateRootCommand](command: T)(implicit validator: ValidateableCommand[T]): Future[_] = ???
+  }
+
+  val cargoRoute = new CargoRoute(commandGateway, cargoQueries)
+
+    "The Cargo route" should "fetch the data of a specific piece of cargo" in {
+      Get(s"/cargo/${cargoId1.id}") ~> Route.seal(cargoRoute.routes) ~> check {
+        status must be(StatusCodes.OK)
```

Comment:
If you test fetching specific data, you should validate the specific data is actually retrieved.

Reply:
couldn't get the marshaling working.. so if you know how todo that :)

Reply:
there should be `SprayJsonSupport` in scope for that to work IIRC